src/WebJobs.Script.WebHost/Diagnostics/IWebHostLoggerProviderFactory.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public interface IWebHostLoggerProviderFactory : ILoggerProviderFactory
+    {
+        IEnumerable<ILoggerProvider> CreateFallbackLoggerProviders();
```

Comment:
I don't like this 'Fallback' concept, but this will all change pretty dramatically when we fully integrate everything with the DI pipeline. In the interest of time, I wanted to get it working similarly to how it works today with TraceWriter before changing it. I'll work with @fabiocav  to make sure this is all addressed when the DI work lands..

Reply:
Katy has the same issue for https://github.com/Azure/azure-webjobs-sdk-script/issues/2186. We actually need a solution ASAP in the v1.x branch for ensuring there is always a default (or "fallback") logger available.

Can't we do something like I suggest in that issue - just make sure we have a default logger configured. When we spin up a host, we use the default until after we read config settings and can configure filtered loggers.

Reply:
Yeah, let me simplify this. I was trying to make it as pluggable as possible (i.e. if the CLI wanted to support a ConsoleLogger as the fallback) -- but I'm not sure that's ever going to be needed. With the DI changes coming, this will probably be re-factored again anyway, so let's keep it simple.